csp_billing_adapter/bill_utils.py

```diff
@@ -429,7 +429,7 @@ def process_metering(
             billed_dimensions
         )
 
-        record_id = retry_on_exception(
+        record_ids = retry_on_exception(
```

Comment:
So given that existing hook implementations may be returning a string rather than a list, it might be was to put in some backwards compatibility support after this meter_billing() hook call along the lines of:

```python
        if type(record_ids) is str:
            record_ids = [record_ids]
```

This checks if the return value from the hook call is a string, and if so converts it into a list containing that string as it's sole element.